data/index.js

```diff
@@ -102,33 +117,136 @@ export function select( reducerKey ) {
 }
 
 /**
- * Higher Order Component used to inject data using the registered selectors.
+ * Returns the available actions for a part of the state.
  *
- * @param {Function} mapSelectorsToProps Gets called with the selectors object
- *                                       to determine the data for the
- *                                       component.
+ * @param {string} reducerKey Part of the state shape to dispatch the
+ *                            action for.
  *
- * @return {Function} Renders the wrapped component and passes it data.
+ * @return {*} The action's returned value.
  */
-export const query = ( mapSelectorsToProps ) => ( WrappedComponent ) => {
-	const store = {
-		getState() {
-			return mapValues( stores, subStore => subStore.getState() );
-		},
-		subscribe,
-		dispatch() {
-			// eslint-disable-next-line no-console
-			console.warn( 'Dispatch is not supported.' );
-		},
-	};
-	const connectWithStore = ( ...args ) => {
-		const ConnectedWrappedComponent = connect( ...args )( WrappedComponent );
-		return ( props ) => {
-			return <ConnectedWrappedComponent { ...props } store={ store } />;
-		};
-	};
+export function dispatch( reducerKey ) {
+	return actions[ reducerKey ];
+}
+
+/**
+ * Higher-order component used to inject state-derived props using registered
+ * selectors.
+ *
+ * @param {Function} mapStateToProps Function called on every state change,
+ *                                   expected to return object of props to
+ *                                   merge with the component's own props.
+ *
+ * @return {Component} Enhanced component with merged state data props.
+ */
+export const withSelect = ( mapStateToProps ) => ( WrappedComponent ) => {
+	class ComponentWithSelect extends Component {
+		constructor() {
+			super( ...arguments );
+
+			this.runSelection = this.runSelection.bind( this );
+
+			this.state = {};
+		}
+
+		componentWillMount() {
+			this.subscribe();
+
+			// Populate initial state.
+			this.runSelection();
+		}
```

Comment:
Should we add `componentDidUpdate` for use case where the props changes but no subscribe is run?

Reply:
>Should we add `componentDidUpdate` for use-cases where the props changes but no subscribe is run?

I think it would be `componentWillReceiveProps`, but yes, I think so.

I'd also also it would only need to be run if the props have changed.